core/src/net/buffer.rs

```diff
@@ -0,0 +1,662 @@
+use crate::net::{buffer_pool::BufferPool, frames, frames::*};
+use bytes::{Buf, BufMut};
+use core::{cmp, mem, ptr};
+use iovec::IoVec;
+use std::{
+    borrow::Borrow,
+    mem::MaybeUninit,
+    pin::Pin,
+    sync::Arc,
+};
+
+const FRAME_HEAD_LEN: usize = frames::FRAME_HEAD_LEN as usize;
+const BUFFER_SIZE: usize = ENCODEBUFFER_MIN_REMAINING * 1000;
+// Assume 64 byte cache lines -> 1000 cache lines per chunk
+const ENCODEBUFFER_MIN_REMAINING: usize = 64; // Always have at least a full cache line available
+
+/// Required methods for a Chunk
+pub trait Chunk: Send {
+    /// Returns a mutable pointer to the underlying buffer
+    fn as_mut_ptr(&mut self) -> *mut u8;
+    /// Returns the length of the chunk
+    fn len(&self) -> usize;
+}
+
+/// A Default Kompact Chunk
+pub(crate) struct DefaultChunk {
+    chunk: Pin<Box<[u8; BUFFER_SIZE]>>,
```

Comment:
My Pin-foo is not too strong, but I'm like 90% sure that a type like `Pin<Box<[u8; BUFFER_SIZE]>>` doesn't really make sense. I think you can just use `Box<[u8; BUFFER_SIZE]>`. `Box<T>` is `Unpin`, so you can move it anyway, even when it's inside a `Pin`. I think you'd have to have some kind of self-referential type for it to make sense to put it behind a `Pin`.